source/nielvisiii/academicIO.py

```diff
@@ -787,13 +789,15 @@ def __init__(self, frequency,
         """
         super(SPI, self).__init__()
         assert 40 <= frequency <= 4000000
+        assert DIOChannel.DIO0 <= cs <= DIOChannel.DIO19
+        assert type(cs_value) == bool
         assert 4 <= frame_length <= 16
         assert bank in Bank
         assert clock_phase in SPIClockPhase
         assert clock_polarity in SPIClockPolarity
         assert data_direction in SPIDataDirection
-        with DigitalInputOutput(bank, [DIOChannel.DIO0]) as dio:
-            dio.write(False, [DIOChannel.DIO0])
+        with DigitalInputOutput(bank, [cs]) as dio:
```

Comment:
Move the CS related logic completely out of SPI APIs in academicIO.py. Move it into SPI.py example.